### PR TITLE
test: Stripe webhook integration

### PR DESCRIPTION
**Pull request by zodman**
- update
- Integration with stripe webhooks

## What does this PR do?

Finishing the e2e Integration with stripe

Fixes # (issue)

/claim #9366
/claim #9369 
/claim #9370
/claim #9367 

## Requirement/Documentation

For run stripe needs stripe-cli for receive webhooks

## Type of change

  - [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by ologbonowiwi** [comment]
Is there only one way to make the webhook work with the CLI? It is what I'm struggling with on my attempt for #9370.

The current approach could work locally but still needs manual handling for webhooks, which isn't the best approach for automated pipelines that won't run with human intervention. You'll be able to see it on my PR (which I wish I could open a draft now, but I'm unable to access my PC unfortunately)

**Comment by ologbonowiwi** [comment]
We should search for another way to trigger the webhook on Stripe integration; this is what I'm struggling with on my attempt for #9370.

The current approach could work locally but still needs manual handling for webhooks, which isn't the best approach for automated pipelines that won't run with human intervention. You'll be able to see it on my PR (which I wish I could open a draft now, but I'm unable to access my PC, unfortunately)

File: yarn.lock
Comment (ologbonowiwi) [inline yarn.lock:1]: These changes are necessary in the context of these new e2e tests?

**Comment by zodman** [comment]
> We should search for another way to trigger the webhook on Stripe integration; this is what I'm struggling with on my attempt for #9370.
> 
> The current approach could work locally but still needs manual handling for webhooks, which isn't the best approach for automated pipelines that won't run with human intervention. You'll be able to see it on my PR (which I wish I could open a draft now, but I'm unable to access my PC, unfortunately)

No manual interactions needed. If you take a look `e2e.yaml` I have  automated it.

in resume, you need download the stripe-cli, update the STRIPE_WEBHOOK_SECRET and run the e2e in parallel.

```
# you need to update the webhook_secret for getting redirected to localhost (valid for 90 days)
export STRIPE_WEBHOOK_SECRET=$(stripe listen --api-key $STRIPE_PRIVATE_KEY  --print-secret)
npx -y  npm-run-all -r -p -y -l  "stripe:listen" "e2e stripe"
```

the npm-rull-all will fire in parallel with the stripe listen to forward the webhooks to localhost and execute the e2e for stripe.

this was the easiest approach to use and it should work on the GitHub actions without manual interaction.

**Comment by ologbonowiwi** [comment]
> No manual interactions needed. If you take a look e2e.yaml I have automated it.
>
> in resume, you need download the stripe-cli, update the STRIPE_WEBHOOK_SECRET and run the e2e in parallel.
>
> ```
> # you need to update the webhook_secret for getting redirected to localhost (valid for 90 days)
> export STRIPE_WEBHOOK_SECRET=$(stripe listen --api-key $STRIPE_PRIVATE_KEY  --print-secret)
> npx -y  npm-run-all -r -p -y -l  "stripe:listen" "e2e stripe"
> ```
> the npm-rull-all will fire in parallel with the stripe listen to forward the webhooks to localhost and execute the e2e for stripe.
>
> this was the easiest approach to use and it should work on the GitHub actions without manual interaction.

But where are we doing the trigger to the paid event? Didn't found it on the PR

**Comment by zodman** [comment]
> But where are we doing the trigger to the paid event? Didn't found it on the PR
> > No manual interactions needed. If you take a look e2e.yaml I have automated it.
> > in resume, you need download the stripe-cli, update the STRIPE_WEBHOOK_SECRET and run the e2e in parallel.
> > ```
> > # you need to update the webhook_secret for getting redirected to localhost (valid for 90 days)
> > export STRIPE_WEBHOOK_SECRET=$(stripe listen --api-key $STRIPE_PRIVATE_KEY  --print-secret)
> > npx -y  npm-run-all -r -p -y -l  "stripe:listen" "e2e stripe"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > the npm-rull-all will fire in parallel with the stripe listen to forward the webhooks to localhost and execute the e2e for stripe.
> > this was the easiest approach to use and it should work on the GitHub actions without manual interaction.
> 
> But where are we doing the trigger to the paid event? Didn't found it on the PR

after the user introduces the CC and clicks pay. In the stripes servers send a webhook to the device registered by the `stripe listen`  what is received forwarded to  `$NEXT_PUBLIC_WEBAPP_URL/api/integrations/stripepayment/webhook` without this `stripe listen` you will never receive the webhook.

**Comment by ologbonowiwi** [comment]
> after the user introduces the CC and clicks pay. In the stripes servers send a webhook to the device registered by the stripe listen what is received is forwarded to $NEXT_PUBLIC_WEBAPP_URL/api/integrations/stripepayment/webhook without this stripe listen, you will never receive the webhook.

You're right, but your code still doesn't hit the webhook with the booking paid event. For example, in the "Paid booking should be able to be cancelled and refunded" test case, you should be validating the message "This booking payment has been refunded." on the screen, which probably does not appear since the paid trigger was never hit.

I successfully fixed this, if you agree to split the bounties I can help you on the implementation

**Comment by ologbonowiwi** [comment]
Hey @zodman, the `BOOKING_PAID` webhook it's already implemented on the project; but your test case doesn't validate it 👀

**Review by joeauyeung — CHANGES_REQUESTED** [review]
Thanks for your contribution @zodman. I left a comment about using the `stripe.webhooks.generateTestHeaderString` method.

**Pull request closed without merging**